### ISSUE TITLE
1.3.1 Failed on lower Node.js

### ISSUE THREAD

**Issue by XadillaX** [isaacs/minizlib#9]
because of:

```js
const { ZLIB_VERNUM } = ...;
```

I think this is a breaking change and should increase a major version.

**Comment by czy88840616** [comment]
same problem

**Comment by fengmk2** [comment]
https://github.com/isaacs/minizlib/issues/10 same

**Comment by jasonzhuang** [comment]
same

**Comment by yxu2018** [comment]
same

**Comment by chemitaxis** [comment]
same here... :(

**Comment by molinto** [comment]
Same here, please revert - cannot build :(

**Comment by molinto** [comment]
```
npm link bcrypt

> bcrypt@3.0.6 install /usr/lib/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using needle for node-pre-gyp https download 
node-pre-gyp ERR! UNCAUGHT EXCEPTION 
node-pre-gyp ERR! stack TypeError: Cannot match against 'undefined' or 'null'.
node-pre-gyp ERR! stack     at Object.<anonymous> (/usr/lib/node_modules/bcrypt/node_modules/minizlib/constants.js:2:40)
node-pre-gyp ERR! stack     at Module._compile (module.js:577:32)
node-pre-gyp ERR! stack     at Object.Module._extensions..js (module.js:586:10)
node-pre-gyp ERR! stack     at Module.load (module.js:494:32)
node-pre-gyp ERR! stack     at tryModuleLoad (module.js:453:12)
node-pre-gyp ERR! stack     at Function.Module._load (module.js:445:3)
node-pre-gyp ERR! stack     at Module.require (module.js:504:17)
node-pre-gyp ERR! stack     at require (internal/module.js:20:19)
node-pre-gyp ERR! stack     at Object.<anonymous> (/usr/lib/node_modules/bcrypt/node_modules/minizlib/index.js:7:39)
node-pre-gyp ERR! stack     at Module._compile (module.js:577:32)
node-pre-gyp ERR! System Linux 4.15.0-55-generic
node-pre-gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/bcrypt/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/lib/node_modules/bcrypt
node-pre-gyp ERR! node -v v6.17.1
node-pre-gyp ERR! node-pre-gyp -v v0.12.0
node-pre-gyp ERR! This is a bug in `node-pre-gyp`.
node-pre-gyp ERR! Try to update node-pre-gyp and file an issue if it does not help:
node-pre-gyp ERR!     <https://github.com/mapbox/node-pre-gyp/issues>
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! bcrypt@3.0.6 install: `node-pre-gyp install --fallback-to-build`
```

@isaacs I have to use Node 6.9.5 as have hardware & cannot update.

**Comment by AliF50** [comment]
We have the same issue on node version 6.10.0.

**Comment by himanshusinghs** [comment]
For those still looking for solution, please follow the other commits referenced above. 
In our case we were using -
`node`: 6.10.0
`npm`: 3.10.10
`bcrypt`: 3.0.5 which required `minizlib`. So we added `minizlib`'s `v1.2.2` as a dependency to our package.json and things are smooth again.

There's a better approach apparently, by specifying `resolutions` in package.json and locking the version of minizlib to "v1.2.2" in there but we tried and it was futile.

**Comment by molinto** [comment]
Hey, @himanshusinghs, just tried adding "minizlib": "1.2.2" to out package.json, but same result on Gitlab :(

**Comment by Srokap** [comment]
@molinto check _requiredBy entries in package.json of minizlib to see what requires it. Tar module seems to allow version 1.2.2 and adding it to main file as exact version worked for me.

**Comment by molinto** [comment]
@Srokap Cannot see any required in package.json (https://github.com/isaacs/minizlib/blob/master/package.json)

or v1.2.2:
https://github.com/isaacs/minizlib/blob/199410f31c7537f85712bf82d31746090d313d1c/package.json

Where am I looking please?

**Comment by himanshusinghs** [comment]
@Srokap you probably wanted to mention `package-lock.json` ?

**Comment by isaacs** [comment]
Fixed on 1.3.2.

**Comment by Srokap** [comment]
@himanshusinghs I am on node 6 with bundled npm, so no package locks for me. I can check it by running:
```
cat node_modules/minizlib/package.json | jq '._requiredBy'
```
I found that tar module was requiring `^1.2.2` version which allowed to make it use older version when required at root level.
